### PR TITLE
Added AnimeUnity Italian Server

### PR DESCRIPTION
**Pull request by Catta1997**
Added Italian server

### CODE REVIEW

**Comment by Catta1997** [comment]
Can you update TestFlight beta with this?

File: NineAnimator/Models/Anime Source/animeunity.it/AnimeUnity+Anime.swift [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                var episodeLink = try container.select("a").attr("href")
+                print(episodeLink)
+                episodeLink = episodeLink.replacingOccurrences(of: "'", with: "\'")
+                print(episodeLink)
+                print("-----------")
+                if (!episodeLink.isEmpty){
+                    print(episodeLink)
+                    let newString = episodeLink.replacingOccurrences(of: "\'", with: "%27")
+                    print("--")
+                    print(newString)
```
Comment (SuperMarcus) [inline NineAnimator/Models/Anime Source/animeunity.it/AnimeUnity+Anime.swift]:
```suggestion
                episodeLink = episodeLink.replacingOccurrences(of: "'", with: "\'")
                if (!episodeLink.isEmpty){
                    let newString = episodeLink.replacingOccurrences(of: "\'", with: "%27")
```

Can you remove the debugging prints or use `Log.debug()` instead?

File: NineAnimator/Models/Anime Source/animeunity.it/AnimeUnity+Anime.swift [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                if (try entry.text().contains("TRAMA")){
+                    let trama = try entry.text()
+                    animeSynopsis = String(trama.dropFirst(7))
+//                     = try entry.text()
+                }
+                                if (try entry.text().contains("TITOLO")){
+                                    let title = try entry.text()
+                                    animeTitle = String(title.dropFirst(8))
+                //                     = try entry.text()
+                                }
```
Comment (SuperMarcus) [inline NineAnimator/Models/Anime Source/animeunity.it/AnimeUnity+Anime.swift]: Can you fix the indentations here or run `swiftlint autocorrect`?

File: NineAnimator/Models/Anime Source/animeunity.it/AnimeUnity+Search.swift [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+import Alamofire
+import Foundation
+import SwiftSoup
+
+extension NASourceAnimeUnity {
+    class SearchAgent: ContentProvider {
+            var totalPages: Int? { 1 }
+            var availablePages: Int { _results == nil ? 0 : 1 }
+            var moreAvailable: Bool { _results == nil }
+            var stringa:String!
```
Comment (SuperMarcus) [inline NineAnimator/Models/Anime Source/animeunity.it/AnimeUnity+Search.swift]: Can you avoid using the force unwrapping?

File: NineAnimator/Models/Anime Source/animeunity.it/AnimeUnity+Featured.swift [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                        image: artworkUrl,
+                        source: self
+                    ))
+                }
+            }
+            */
+            return BasicFeaturedContainer(
+                featured:[],
+                latest: []
+            )
```
Comment (SuperMarcus) [inline NineAnimator/Models/Anime Source/animeunity.it/AnimeUnity+Featured.swift]: Featured page not implemented.

File: NineAnimator/Models/Anime Source/animeunity.it/AnimeUnity.swift [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    #elseif canImport(AppKit)
+    var siteLogo: NSImage { #imageLiteral(resourceName: "AnimeUnity Site Logo") }
+    #endif
+    
+    var siteDescription: String {
+        "AnimeUnity è un server italiano"
+    }
+    
+    class var FourAnimeStream: Anime.ServerIdentifier {
+        "4anime"
```
Comment (SuperMarcus) [inline NineAnimator/Models/Anime Source/animeunity.it/AnimeUnity.swift]: Why are there references to 4anime?

**Comment by Catta1997** [comment]
how can I upload the corrected file? I have to do a new pull?

**Comment by Catta1997** [comment]
I should have done all the changes required. Also fixed name of file and added the "airdate"

**Comment by SuperMarcus** [comment]
Gotcha. I will review the changes tomorrow. Thanks!

**Comment by Catta1997** [comment]
I noticed that the search sometimes fails and returns "nil" I have tried to fix it but with no success. Does it always cause you problems? Have you tried to use "simulate location" on Italy? maybe the server accept only request from Italy

Here there is a pic of search in my simulator
Schermata 2020-03-22 alle 19 03 06

**Comment by SuperMarcus** [comment]
It's highly likely that the server is only accepting requests from Italy. Simulating locations likely won't help since its for device locations, which NineAnimator requests will not include. However, I am using a VPN to proxy my traffics through an Italian IP address.

The search hasn't worked for me. I've also noticed that the featured page sometimes won't load as well. However, the `/livesearch.php` API appears to be very consistent.

```sh
$ curl -vL -X POST 'https://www.animeunity.it/inc/livesearch.php' --data 'search=naruto'
```

**Comment by Catta1997** [comment]
ok, i will rewrite the search with that url tomorrow. thanks!

**Comment by Catta1997** [comment]
I have tried to implement `/livesearch.php` but return 
`{"status":"failed","message":"No post req."}`
Because
```
requestTask = parent.request(
                    browsePath: "/inc/livesearch.php",
                    query: [ "search": title ]
                )
```
only do a GET request.

Same problem I have with:
```
requestTask = parent.request(
                    browsePath: "/anime.php?c=archive",
                    query: [ "query": title ]
                )
```
If I use:
```
let params: Parameters = ["query": title]
let url = "https://animeunity.it/anime.php?c=archive"
_ = AF.request(url, method: .post, parameters: params)
```
 in a external function to perform a "POST" sometimes return nil

**Comment by Catta1997** [comment]
I have corrected the search and now it won't (or shouldn't) return an empty string @SuperMarcus can you check if it works for you too?

**Pull request merged**